modules/internal/ChapelReduce.chpl

```diff
@@ -52,7 +52,16 @@ module ChapelReduce {
   
   proc chpl__sumType(type eltType) type {
     var x: eltType;
-    return (x + x).type;
+    if isArray(x) {
+      type xET = x.eltType;
+      type xST = chpl__sumType(xET);
+      if xET == xST then
+        return eltType;
+      else
+        return [x.domain] xST;
+    } else {
+      return (x + x).type;
+    }
```

Comment:
It concerns me that this code only applies to += reductions. Why? Shouldn't this work for a *= going through an array? Or anything else?
Why did you need to change this function at all? I would expect (x+x).type to work when x is an array.